qubes/app.py

```diff
@@ -1621,7 +1621,11 @@ def _domain_event_callback(self, _conn, domain, event, _detail, _opaque):
                 )
         elif event == libvirt.VIR_DOMAIN_EVENT_RESUMED:
             try:
-                vm.fire_event("domain-unpaused")
+                if not getattr(vm, "unpaused_fired", False):
```

Comment:
You set the attribute to `False` before calling libvirt resume, which means here it may still be `False` depending on when exactly this event will be handled (can be for example in the middle of `domain-unpaused` handling already). In which case you'll get duplicated events.

Better to reverse the condition - set some variable signaling it was qubesd calling resume, to not fire the event here, and clear that variable here after skipping firing the event.